doc/sampling.md

```diff
@@ -120,5 +120,37 @@ This configuration option has the same meaning as the `DD_TRACE_RATE_LIMIT`
 environment variable.  Note that the environment variable overrides the
 `TracerOptions` field if both are specified.
 
+Span Sampling
+-------------
+Span sampling is used to select spans to keep even when the enclosing
+trace is dropped.
+
+Similar to _trace_ sampling rules, _span_ sampling rules are configured as a
+JSON array of object, where each object may contain the following properties:
+```
+[{
+    "service": <matches the span's service name, or any if absent>,
+    "name": <matches the span's operation name, or any if absent>,
+    "sample_rate": <the probability of sampling matching spans, or 1.0 if absent>,
+    "max_per_second": <limit in spans sampled by this rule each second, or unlimited if absent>
+}, ...]
+```
+
+The `service` and `name` are glob patterns, where "glob" here means:
+- `*` matches any substring, including the empty string,
+- `?` matches exactly one of any character, and
+- any other character matches exactly one of itself.
+
+Span sampling rules are examined only when the enclosing trace is to be
```

Comment:
Does this need to say "automatically dropped" or is it still applied for manual drop?

Reply:
Span sampling rules apply when the trace is dropped by the tracer for any reason, including a manual drop.

Seems odd, right?  I said drop this, but I also said to sometimes keep part of it when we drop it.

Now I'm thinking "well, it's only the statistical trace sampling decisions that customers want to counter with single span sampling, so let's exclude manually dropped traces."  But then I think "ah, but we no longer consider rule-based (i.e. statistics set by the user, e.g. DD_TRACE_SAMPLE_RATE) sampling decisions 'auto', now they're 'user'."

Reply:
I suppose that once we have sampling mechanism propagation (as we recently merged but have not yet released), we could distinguish between "user said drop" and "user said less than 100% and we're being dropped."  Even then, though, we couldn't count on a service sending us a sampling mechanism alongside the sampling priority.